src/pal/src/misc/dbgmsg.cpp

```diff
@@ -470,66 +470,82 @@ Notes :
 int DBG_printf(DBG_CHANNEL_ID channel, DBG_LEVEL_ID level, BOOL bHeader,
                LPCSTR function, LPCSTR file, INT line, LPCSTR format, ...)
 {
-    CHAR *buffer = (CHAR*)alloca(DBG_BUFFER_SIZE);
-    CHAR indent[MAX_NESTING+1];
-    LPSTR buffer_ptr;
-    INT output_size;
-    va_list args;
-    void *thread_id;
-    int old_errno = 0;
-
-    old_errno = errno;
+    struct ErrnoHolder
+    {
+        int value;
+        ErrnoHolder() : value(errno) { }
+        ~ErrnoHolder()
+        {
+            if ( value != errno )
+                errno = value;
+        }
+    } errno_holder;
 
+    CHAR indent[MAX_NESTING+1];
     if(!DBG_get_indent(level, format, indent))
     {
+        // Note: we will drop log messages here if the indent gets too high, and we won't print
```

Comment:
Should it instead just emit at max nesting?

Reply:
I mean, when you have to resort to look at debug output, you're probably going to be glad for whatever information you can get, even if it's not pretty.

Reply:
It looks like in some cases `DBG_get_indent` will print an error but not in all.  Here where we call the function, we don't know if it failed because the indent was to high or for some other reason.  Maybe  a good solution would be to have `DBG_get_indent` to print an error in all cases...